clearwater-config-manager.root/usr/share/clearwater/clearwater-config-manager/scripts/confirm_process.py

```diff
@@ -0,0 +1,24 @@
+
+# Copyright (C) Metaswitch Networks.
+"""Script to ask the user to confirm a process"""
+
+
+def Confirm_yn(prompt):
+    """asks the user to confirm they want to make the changes described by the
+    prompt passed in. This keeps asking the user until a valid response is
+    given."""
+    question = "Do you want to continue?"
+    allowed = False
+    while allowed is False:
+        print '\n{0} '.format(prompt)
+        supplied_input = raw_input('{0} [yes/no] '.format(question))
+        try:
+            if supplied_input.strip().lower() not in ['y', 'yes', 'n', 'no']:
+                raise ValueError('Answer must be yes or no')
+            else:
+                response = supplied_input.lower().startswith('y')
+                allowed = True
+        except ValueError as exc:
```

Comment:
This is quite a heavyweight way of determining whether the supplied answer is valid. I think you could write this code in a neater way without using exceptions - the result of `supplied_input.strip().lower() not in ['y', 'yes', 'n', 'no']` contains all the information you need to know whether or not a valid answer is present, so why create a second mechanism for reporting the error?

Reply:
Try and re-write this without the `try/except` blocks. Let me know if you need any pointers.